logger.go

```diff
@@ -109,22 +103,61 @@ func createSentryHook(sentryDsn string) logrus.Hook {
 }
 
 /*
+getLevelFromEnv parses log level from environment variable LOGGER_LEVEL and apply sensible defaults when the value is
+absent or invalid.
+
+When LOGGER_LEVEL is not defined we use DebugLevel, which is convenient for development
+
+When LOGGER_LEVEL is invalid we use InfoLevel, which is convenient for production, and log a warning to help fix
+the situation
+
+Note: we do not use logrus.ParseLevel because we want to exclude warn, fatal and panic which are not a part of cp
+conventions, and we need to have error messages consistent with what's actually possible.
+ */
+func getLevelFromEnv() (logrus.Level, error) {
+	levelStr := os.Getenv("LOGGER_LEVEL")
+	if levelStr == "" {
+		return logrus.DebugLevel, nil
+	}
+
+	switch strings.ToLower(levelStr) {
+	case "error":
+		return logrus.ErrorLevel, nil
+	case "warning":
+		return logrus.WarnLevel, nil
+	case "info":
+		return logrus.InfoLevel, nil
+	case "debug":
+		return logrus.DebugLevel, nil
+	}
+
+	return logrus.InfoLevel, fmt.Errorf("not a valid logrus Level: %q", levelStr)
+}
+
+/*
 CreateLogger creates a new instance of logrus.Logger, which is configured from the environment variables according to cp
 conventions (see package overview)
  */
 func CreateLogger() *logrus.Logger {
+	level, levelParseErr := getLevelFromEnv()
```

Comment:
I can't do it yet, because the error handling consists in logging a warning, so we need the logger initialization (and especially the sentry hook) to be complete before logging the warning. I will add a comment to explain this.